src/WingmanDataTrail/MetricVizPanel/WithUsageDataPreviewPanel.tsx

```diff
@@ -147,33 +154,80 @@ export class WithUsageDataPreviewPanel extends SceneObjectBase<WithUsageDataPrev
           singularUsageType={usageDetails[sortBy].singularUsageType}
           pluralUsageType={usageDetails[sortBy].pluralUsageType}
           icon={usageDetails[sortBy].icon as IconName}
+          usageDetails={usageDetails[sortBy].usageDetails}
         />
       </div>
     );
   };
 }
-
+// TODO: pass full MetricUsageDetailsObject into UsageSectionProps?
 interface UsageSectionProps {
   usageType: MetricUsageType;
   usageCount: number;
   singularUsageType: string;
   pluralUsageType: string;
   icon: IconName;
+  usageDetails: MetricUsageDetails;
 }
 
-function UsageData({ usageType, usageCount, singularUsageType, pluralUsageType, icon }: Readonly<UsageSectionProps>) {
+function UsageData({
+  usageType,
+  usageCount,
+  singularUsageType,
+  pluralUsageType,
+  icon,
+  usageDetails,
+}: Readonly<UsageSectionProps>) {
   const styles = useStyles2(getStyles);
 
+  let dashboardItems: Array<{ label: string; value: string; count: number }> = [];
+  if (usageDetails.usageType === 'dashboard-usage') {
+    const { dashboards } = usageDetails;
+    dashboardItems = Object.entries(dashboards)
+      .map(([name, dashboardInfo]) => ({
+        label: `${name.length > 25 ? name.substring(0, 22) + '...' : name} (${dashboardInfo.count})`, // truncate long dashboard names while preserving the count
+        value: `/d/${dashboardInfo.uid}`,
+        count: dashboardInfo.count,
+      }))
+      .sort((a, b) => b.count - a.count);
+  }
+
   return (
     <div className={styles.usageContainer} data-testid="usage-data-panel">
-      <Tooltip
-        content={`Metric is used in ${usageCount} ${usageCount === 1 ? singularUsageType : pluralUsageType}`}
-        placement="top"
-      >
-        <span className={styles.usageItem} data-testid={usageType}>
-          <Icon name={icon} /> {usageCount}
-        </span>
-      </Tooltip>
+      {usageDetails.usageType === 'dashboard-usage' ? (
+        <>
+          <Dropdown
+            placement="right-start"
+            overlay={
+              <Menu style={{ maxHeight: '200px', overflowY: 'auto' }}>
```

Comment:
```suggestion
              <Menu style={{ maxHeight: '212px', overflowY: 'auto' }}>
```

SuperNit: 212px to give a chance for the last item to be chopped, which helps understanding there might be more items in the menu:
<img width="259" height="231" alt="image" src="https://github.com/user-attachments/assets/0fa48cc2-cacc-4ec4-b336-d0ce91c1bf4d" />